### PR TITLE
Remove the network controller `previousProviderStore`

### PR DESCRIPTION
**Pull request by Gudahtt**
## Explanation

The `previousProvider` network controller state has been replaced with a private internal property. This was only used internally, it did not need to be part of state.

This relates to https://github.com/MetaMask/metamask-extension/issues/18303

## Manual Testing Steps

No functional changes to test

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: app/scripts/controllers/network/network-controller.test.ts [hunk 1]
```diff
@@ -4137,16 +4127,43 @@ describe('NetworkController', () => {
           },
         },
         async ({ controller }) => {
-          const network = new CustomNetworkCommunications({
```
Comment (mcmire) [inline app/scripts/controllers/network/network-controller.test.ts:4140]: The new version of this test now seems very similar to existing tests for `rollbackToPreviousProvider` (e.g. line 5865). Perhaps we don't need this test anymore?

File: app/scripts/controllers/network/network-controller.test.ts [hunk 2]
```diff
@@ -4639,8 +4656,11 @@ describe('NetworkController', () => {
     for (const { networkType, chainId, ticker } of INFURA_NETWORKS) {
       describe(`given a type of "${networkType}"`, () => {
         it('stores the current provider configuration before overwriting it', async () => {
+          const { unrestrictedMessenger, restrictedMessenger } =
```
Comment (mcmire) [inline app/scripts/controllers/network/network-controller.test.ts:4659]: Similar as above — this one is now similar to the test on line 6508.

File: app/scripts/controllers/network/network-controller.test.ts (hunk 1 above)
Comment (Gudahtt) [reply app/scripts/controllers/network/network-controller.test.ts:4140]: Good point. Yeah that would make sense to me, I'll remove both of these that you've commented on.

Comment (Gudahtt) [reply app/scripts/controllers/network/network-controller.test.ts:4140]: Removed in 2ea6a0476a7634ce97d4bef938ed7c1fc5198a67

File: app/scripts/controllers/network/network-controller.test.ts (hunk 2 above)
Comment (Gudahtt) [reply app/scripts/controllers/network/network-controller.test.ts]: Removed in 2ea6a0476a7634ce97d4bef938ed7c1fc5198a67

**Review by mcmire — APPROVED** [review]
LGTM!

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**